Client/src/Validation/validation.js

```diff
@@ -128,10 +128,57 @@ const maintenanceWindowValidation = joi.object({
 	monitors: joi.array().min(1),
 });
 
+const advancedSettingsValidation = joi.object({
+	apiBaseUrl: joi.string().uri({ allowRelative: true }).trim().messages({
+		"string.empty": "API base url is required.",
+		"string.uri": "The URL you provided is not valid.",
+	}),
+	logLevel: joi.string().valid("debug", "none", "error", "warn").allow(""),
+	systemEmailHost: joi.string().allow(""),
+	systemEmailPort: joi.string().allow(""),
+	systemEmailAddress: joi.string().allow(""),
+	systemEmailPassword: joi.string().allow(""),
+	jwtTTL: joi
+		.string()
+		.trim()
+		.messages({
+			"string.empty": "JWT TTL is required.",
+		})
+		.custom((value, helpers) => {
+			// test against regx start with a number followed by number or .,
+			//ends with d or h
+			const regex = /^[1-9]+\d*(.)*(d|h)$/;
+			const found = value.match(regex);
+			if (!found) {
+				return helpers.message(
+					"JWT TTL should start with a non zero number and ends with 'd' or 'h'."
+				);
```

Comment:
I guess the validation can be done on the combined string with nunber and unit?

The regex is an issue though, I think this will allow for strings like "123abcd" which is not what is intended I don't think.  If you drop the `(.)` I think you're good to go.

You can test the [regex here](https://regex101.com/)

Reply:
resolved in :https://github.com/bluewave-labs/bluewave-uptime/pull/1020/commits/57edff36c9920fc569f24c7c11dab21f81704b12